runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java

```diff
@@ -331,6 +332,12 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		int itemId = event.getIdentifier();
 		String option = Text.removeTags(event.getOption()).toLowerCase();
 		String target = Text.removeTags(event.getTarget()).toLowerCase();
+		NPC hintArrowNpc  = client.getHintArrowNpc();
+
+		if (hintArrowNpc != null && hintArrowNpc.getName().toLowerCase().equals(target))
```

Comment:
You could probably compare ID instead of name

Reply:
done